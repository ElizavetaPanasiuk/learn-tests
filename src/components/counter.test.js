import { configure, shallow } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";

import Counter from './Counter';

configure({
  adapter: new Adapter()
});

describe('Count component', () => {
  let component = shallow(<Counter />);

  it('should render counter component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Counter handlers', () => {
    it('should changeout value to plus one', () => {
      const btn = component.find('.plusOneBtn');
      btn.simulate('click');
      expect(toJson(component)).toMatchSnapshot();
      expect(component.state().count).toBe(1);
    })
  })
});
