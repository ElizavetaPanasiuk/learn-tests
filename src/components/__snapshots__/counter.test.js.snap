// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count component Counter handlers should changeout value to plus one 1`] = `
<div>
  <div>
    1
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;

exports[`Count component should render counter component 1`] = `
<div>
  <div>
    0
  </div>
  <button
    className="plusOneBtn"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="resetBtn"
    onClick={[Function]}
  >
    Reset to 10
  </button>
</div>
`;
